src/provider/index.ts

```diff
@@ -1,132 +1,114 @@
-import {AppId, Application, AppName} from '../client/directory';
-import {IntentType} from '../client/intents';
-import {RaiseIntentPayload} from '../client/internal';
-import {Context} from '../client/main';
-
-import {FDC3} from './FDC3';
-import {AppMetadata} from './MetadataStore';
-
-console.log('the provider has landed.');
-
-// Create and initialise desktop agent
-const service: FDC3 = new FDC3();
-service.register();
-
-/**
- * When the user is shown an app selector popup, they have the option of telling
- * the service how to handle similar intents in the future.
- *
- * This enum defines the options available to users.
- */
-export const enum DefaultAction {
-    /**
-     * Service should always show the app selection UI, to allow the user to
-     * choose which application to use.
-     */
-    ALWAYS_ASK = 'ALWAYS_ASK',
-
-    /**
-     * The service should always use the current selection when the intent is
-     * coming from the app that fired the current intent.
-     */
-    ALWAYS_FOR_APP = 'ALWAYS_FOR_APP',
-
-    /**
-     * The service should always use the current selection, whenever an intent of
-     * this type is fired.
-     */
-    ALWAYS_FOR_INTENT = 'ALWAYS_FOR_INTENT'
+import 'reflect-metadata';
+
+import {RaiseIntentPayload, APITopic, OpenPayload, FindIntentPayload, FindIntentsByContextPayload, BroadcastPayload} from '../client/internal';
+import {AppIntent, IntentResolution, Application, Intent} from '../client/main';
```

Comment:
Not sure if it's a good idea for provider/index to reference client/main. Could we split those shared types into a separate file or something? Could maybe have them in a hidden file and re-exported in main to keep the docs nice?

Reply:
What's the concern there? Those four types are all returnable from the API, and so shouldn't be in a hidden file.

No objections to re-arranging some of those types from `main.ts` to a more-specific file.

Reply:
The hidden/re-exporting suggestion was because it would allow them to appear in the top level docs but be imported into the provider without needing to import a client file with executing code, which is the crux of my issue with this pattern.